src/main/java/hudson/plugins/collabnet/orchestrate/BuildNotifier.java

```diff
@@ -122,29 +125,27 @@ public BuildNotifier(OptionalAssociationView associationView,
             this.ctfUrl = null;
             this.setUseAssociationView(false);
         }
-        if(eventQ != null) {
-            this.serverUrl = eventQ.serverUrl;
-            this.serverUsername = eventQ.serverUsername;
-            this.serverPassword = eventQ.serverPassword;
-            this.sourceKey = eventQ.sourceKey;
+        if (config.value.equals("eventQ")) {
+            this.serverUrl = config.serverUrl;
+            this.serverUsername = config.serverUsername;
+            this.serverPassword = config.serverPassword;
+            this.sourceKey = config.sourceKey;
             this.setSupportEventQ(true);
-        }
-        if(webhook != null){
-            this.webhookUrl = webhook.webhookUrl;
-            this.webhookUsername = webhook.webhookUsername;
-            this.webhookPassword = webhook.webhookPassword;
-            this.setSupportWebhook(true);
+        } else if (config.value.equals("webhook")) {
+            try {
+                this.webhookUrl = config.getWebhookUrl();
+                if (StringUtils.isEmpty(this.webhookUrl)) {
+                    this.webhookUrl = Helper.getWebhookUrl(this.ctfUrl);
+                }
+                config.setWebhookUrl(this.webhookUrl);
```

Comment:
Move this line inside the if block above

Reply:
User will be entering the webhook url in the given input area but if the user miss to enter the webhook url, it will generate the webhook url using the CTF url. Irrespective of both the cases either webhook url is available or not, the value needs to be set. So, this line cannot be moved inside.

Reply:
Based on your code, the URL is generated only if it is null or empty. Otherwise you are setting it back to the same value. So. It can be moved into the if-block.